Stratis.Bitcoin/Wallet/Tracker.cs

```diff
@@ -0,0 +1,63 @@
+﻿using System;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Stratis.Bitcoin.Wallet.Notifications;
+using Microsoft.Extensions.Logging;
+using NBitcoin;
+using Stratis.Bitcoin.Notifications;
+using Stratis.Bitcoin.Utilities;
+
+namespace Stratis.Bitcoin.Wallet
+{
+    public class Tracker : ITracker
+    {
+        private readonly WalletManager walletManager;
+        private readonly ConcurrentChain chain;
+        private readonly Signals signals;
+        private readonly CoinType coinType;
+        private readonly ILogger logger;
+
+
+		public Tracker(ILoggerFactory loggerFactory, IWalletManager walletManager, ConcurrentChain chain, Signals signals, 
+			Network network)
+        {
+            this.walletManager = walletManager as WalletManager;
+            this.chain = chain;
+            this.signals = signals;
+            this.coinType = (CoinType)network.Consensus.CoinType;
+            this.logger = loggerFactory.CreateLogger(this.GetType().FullName);
+        }
+
+        /// <inheritdoc />
+        public void Initialize()
+        {
+
+	        // subscribe to receiving blocks and transactions
+			IDisposable sub = new BlockSubscriber(this.signals.Blocks, new BlockObserver(this.chain, this.walletManager)).Subscribe();
+	        IDisposable txSub = new TransactionSubscriber(this.signals.Transactions, new TransactionObserver(this.walletManager)).Subscribe();
+
+            // start syncing blocks
+            var bestHeightForSyncing = this.FindBestHeightForSyncing();
+            this.logger.LogInformation($"Tracker initialized. wallet at block {bestHeightForSyncing}.");
```

Comment:
As decided we will make the Traker a base SyncManager and that will manage the wallet height, the light wallet will have its own implementation of the SyncManager (the notification)